kv/txn_coord_sender.go

```diff
@@ -788,28 +587,22 @@ func (tc *TxnCoordSender) heartbeat(id string, trace *tracer.Trace, ctx context.
 		return false
 	}
 
-	request := &proto.HeartbeatTxnRequest{
-		RequestHeader: proto.RequestHeader{
-			Key: txn.Key,
-			Txn: &txn,
-		},
-	}
-
-	request.Header().Timestamp = tc.clock.Now()
-	reply := &proto.HeartbeatTxnResponse{}
-	call := proto.Call{
-		Args:  request,
-		Reply: reply,
-	}
+	hb := &proto.HeartbeatTxnRequest{}
```

Comment:
There's only key left in the header (which is the desired end state for requests), and the rest is on the batch. Awkward to spell out the request heart twice for that, hence my preference.